xbmc/utils/GLUtils.cpp

```diff
@@ -259,3 +260,21 @@ int KODI::UTILS::GL::glFormatElementByteCount(GLenum format)
     return 1;
   }
 }
+
+uint8_t KODI::UTILS::GL::GetChannelFromARGB(const KODI::UTILS::GL::ColorChannel colorChannel,
+                                            const uint32_t argb)
+{
+  switch (colorChannel)
+  {
+    case KODI::UTILS::GL::ColorChannel::A:
+      return (argb >> 24) & 0xFF;
+    case KODI::UTILS::GL::ColorChannel::R:
+      return (argb >> 16) & 0xFF;
+    case KODI::UTILS::GL::ColorChannel::G:
+      return (argb >> 8) & 0xFF;
+    case KODI::UTILS::GL::ColorChannel::B:
+      return (argb >> 0) & 0xFF;
+    default:
+      throw std::runtime_error("KODI::UTILS::GL::GetChannelFromARGB: ColorChannel not handled");
+  };
+}
```

Comment:
IMO has no sense make this outside ColorUtils recently has been reorganized to be more extensible, and we still start to fragment the utilities again?
you can take ispiration from `UTILS::COLOR::MakeColorInfo` and add to `ColorInfo` struct the alpha/red/green/blue entries, after that for example this:

```
float r = KODI::UTILS::GL::GetChannelFromARGB(KODI::UTILS::GL::ColorChannel::R, color) / 255.0f;
float g = KODI::UTILS::GL::GetChannelFromARGB(KODI::UTILS::GL::ColorChannel::G, color) / 255.0f;
float b = KODI::UTILS::GL::GetChannelFromARGB(KODI::UTILS::GL::ColorChannel::B, color) / 255.0f;
float a = KODI::UTILS::GL::GetChannelFromARGB(KODI::UTILS::GL::ColorChannel::A, color) / 255.0f;
```

will be something like:
```
UTILS::COLOR::ColorInfo& ci = UTILS::COLOR::MakeColorInfoCh(color);
float r = ci.red / 255.0f;
float g = ci.green / 255.0f;
float b = ci.blue / 255.0f;
float a = ci.alpha / 255.0f;
```

Reply:
Yea, as stated in the OP I thought about using `ColorUtils` but ultimately decided on putting it in `GLUtils` as the conversion is used specifically in GL code paths. I would think that `ColorUtils` would be more generic. If it's really a big concern I can move it though. I'm really just here to get rid of the macros.

I'll let someone else chime in with an opinion before changing anything.

Reply:
As this is highly localized on the rendering subsystem and results into an improvement (removing the macros) I'm fine keeping this on GLUtils. It seems GLUtils is already included so reusing ColorInfo would result in just another include. Furthermore, ColorInfo seems to have a bit of bloat for what is needed here.
Conclusion: using colorinfo would be good for re usability but the world won't end if this is kept as is :)